gluten-flink/ut/src/test/java/org/apache/gluten/streaming/api/operators/GlutenStreamJoinOperatorTestBase.java

```diff
@@ -0,0 +1,288 @@
+            leftInput,
+            rightInput,
+            leftNode,
+            rightNode,
+            outputVeloxType);
+
+    return new GlutenVectorTwoInputOperator(
+        new StatefulPlanNode(join.getId(), join),
+        leftInput.getId(),
+        rightInput.getId(),
+        leftVeloxType,
+        rightVeloxType,
+        Map.of(join.getId(), outputVeloxType));
+  }
+
+  protected void processTestData(
+      TwoInputStreamOperatorTestHarness<StatefulRecord, StatefulRecord, RowData> harness,
+      List<RowData> leftData,
+      List<RowData> rightData)
+      throws Exception {
+    long timestamp = 0L;
+    for (RowData row : leftData) {
+      StatefulRecord record = convertToStatefulRecord(row, leftVeloxType);
+      harness.processElement1(new StreamRecord<>(record, timestamp++));
+    }
+
+    timestamp = 0L;
+    for (RowData row : rightData) {
+      StatefulRecord record = convertToStatefulRecord(row, rightVeloxType);
+      harness.processElement2(new StreamRecord<>(record, timestamp++));
+    }
+  }
+
+  protected List<RowData> extractOutputFromHarness(
+      TwoInputStreamOperatorTestHarness<StatefulRecord, StatefulRecord, RowData> harness) {
+    Queue<Object> outputQueue = harness.getOutput();
+    return outputQueue.stream()
+        .filter(record -> record instanceof StreamRecord)
+        .map(record -> ((StreamRecord<RowData>) record).getValue())
+        .collect(Collectors.toList());
+  }
+
+  private void setupGlutenResources() {
+    sharedMemoryManager = MemoryManager.create(AllocationListener.NOOP);
+    sharedSession = Velox4j.newSession(sharedMemoryManager);
+    sharedAllocator = new RootAllocator(Long.MAX_VALUE);
+  }
+
+  private void cleanupGlutenResources() {
+    try {
+      if (sharedAllocator != null) {
+        sharedAllocator.close();
+        sharedAllocator = null;
+      }
+      if (sharedSession != null) {
+        sharedSession.close();
+        sharedSession = null;
+      }
+      if (sharedMemoryManager != null) {
+        sharedMemoryManager.close();
+        sharedMemoryManager = null;
+      }
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+  }
+
+  private StatefulRecord convertToStatefulRecord(RowData rowData, RowType rowType) {
+    try {
+      RowVector rowVector =
+          FlinkRowToVLVectorConvertor.fromRowData(rowData, sharedAllocator, sharedSession, rowType);
+
+      StatefulRecord record = new StatefulRecord(null, 0, 0, false, -1);
+      record.setRowVector(rowVector);
+
+      return record;
+    } catch (Exception e) {
+      throw new RuntimeException("Failed to convert RowData to StatefulRecord", e);
+    }
+  }
+
+  protected void executeJoinTest(
+      GlutenVectorTwoInputOperator operator,
+      List<RowData> leftData,
+      List<RowData> rightData,
+      List<RowData> expectedOutput)
+      throws Exception {
+    TwoInputStreamOperatorTestHarness<StatefulRecord, StatefulRecord, RowData> harness =
```

Comment:
I note Flink's test is using KeyedTwoInputStreamOperatorTestHarness. It cannot be used here for now?